pkg/controller/operators/operatorcondition_controller_test.go

```diff
@@ -313,6 +313,44 @@ var _ = Describe("OperatorCondition", func() {
 							Value: operatorCondition.GetName(),
 						}))
 					}
+
+					Context("if the OPERATOR_CONDITION_NAME environment variable is removed", func() {
+						deployment := &appsv1.Deployment{}
+						Eventually(func() error {
+							err := k8sClient.Get(ctx, types.NamespacedName{Name: operatorCondition.Spec.Deployments[0], Namespace: namespace.GetName()}, deployment)
+							if err != nil {
+								return err
+							}
+							deployment.Spec.Template.Spec.Containers[0].Env = nil
+							return k8sClient.Update(ctx, deployment)
+						}, timeout, interval).Should(BeNil())
```

Comment:
Should this setup code to delete the env be in a `BeforeEach` block to ensure it runs before each test in this context (even if there is only one now)?